cachelib/cachebench/cache/Cache.h

```diff
@@ -352,6 +361,9 @@ class Cache {
   // tracker for consistency monitoring.
   std::unique_ptr<ValueTracker> valueTracker_;
 
+  // expected value of all items in Cache.
+  std::optional<std::string> expectedValue_;
```

Comment:
This sounds like we are validating the value, but we are only validating a few bytes in each item. So either it is named incorrectly or we are intending to do something, but the code does not do all of it. 

What's the intent with adding this value validation ? My understanding was that this was to read the item's payload in entirety so that we can benchmark any DRAM or PMem bandwidth numbers. If that's the case, why are we reading only a subset of the item ?

Reply:
Your understanding is correct. I believe we are verifying all bytes of the value right now. `expectedValue_` is being set to `hardcodedString_` which is 4MB. `The std::min<size_t>(expected.size(), getSize(it))` part is just to make sure that we are not reading outside of the item. This is analogous to: https://github.com/facebook/CacheLib/blob/main/cachelib/cachebench/cache/Cache-inl.h#L624

Is this reasoning correct?

Reply:
> expectedValue_ is being set to hardcodedString_ which is 4MB.

The value is set by the caller (stressor) in a certain way, and we are leaking that into the cache. The fact that expectedValue[0 - item.getSize() -1] is the expected value is assumed in the cache. 

Taking a step back, we are not really building this feature to check item's content. That functionality is already handled by the consistency checking feature. We want this feature to simply read the item payload. Can this be done by the CacheStressor instead ? We also don't need to compare it against any expected value. As in, the result of that comparison is simply irrelevant. So there is no need to abort if the value is incorrect.

Reply:
You are right that we are not really building this to check item's content. We came up with the idea of using memcmp to make sure that the code which reads the value is not optimized out by the compiler.

I don't think we can easily move the checking to the CacheStressor (all the latency checking code is done inside Cache). However, we can rewrite `validateValue` to not do any comparison. We can just iterate over the value and add a `volatile` somewhere to make sure the code is not optimized out (e.g. calculate the number of zero-bytes in the value and assign this number to a local volatile var). Would this be acceptable?

Reply:
@igchor  yeah, that seems more reasonable,  simpler and we don't need to think about terminating the run based on correctness. There is probably a way to do it without the volatile (check out folly::doNotOptimizeAway).